src/context/AuthContext.js

```diff
@@ -0,0 +1,20 @@
+import { createContext, useContext, useState } from "react";
+const AuthContext = createContext();
+
+const useAuth = () => useContext(AuthContext);
+const AuthContextProvider = ({ children }) => {
+	const [auth, setAuth] = useState(() => {
+		const token = localStorage.getItem("token");
+
+		if (token) return { token, isAuth: true };
+
+		return { token: "", isAuth: false };
```

Comment:
You r setting isauth for maintaining logged in state, but you can do this  by token a well right, like you can toggle and navigate to home page according to token set or not? Any reason , I don't understand that part.

Reply:
Hey yeah actually it is related to local storage, if user logins and refreshes the. page The login state will be lost so to preserve it I saved isAuth and token in local storage and based on it I'm doing but token should not be there in local storage like that and it will be removed later that is why isAuth is needed